### ISSUE TITLE
build scan shows the wrong os version

### ISSUE THREAD

**Issue by alz-ahm** [gradle/gradle#17212]
build scan shows the wrong os version. My os version is `11.2.3 (20D91)` but it shows `Mac OS X 10.16` in the infrastructure section. 

### Context
I was trying to see if a particular issue happens more for Big Sur users by querying our gradle enterprise data.

### Steps to Reproduce 
I also run it on a simple hello world application and I see the same thing, (look at the build scan) so I don't think there is anything project-specific about it. 

### Your Environment
https://scans.gradle.com/s/ler2pffzdwqn2#infrastructure

**Comment by jthurne** [comment]
MacOS Big Sur reports two different version numbers (depending on various criteria). 

With MacOS Big Sur, Apple has decided to go back to incrementing the operating system's major version. Since MacOS 10, Apple has only incremented the minor version number (10.15 was for MacOS Catalina).  Some applications have come to rely on Apple only incrementing the minor version.

For backwards compatibility, Apple decided to give BigSur two version numbers: 11.x (the latest version is 11.3.1, but Big Sur was originally released as 11.0) and 10.16.  Which version is reported to applications and scripts depends on a number of factors.

In the case of Java-based applications, the number reported depends on the version of the XCode SDK the JRE was built against. Currently, most JREs in active use today have been built against XCode before version 12, and so they currently report the OS version as 10.16.

According to [this OpenJDK Bug](https://bugs.openjdk.java.net/browse/JDK-8253702), the OpenJDK-based JREs should be fixed (JREs 16.0.1, 16.0.2, and 11.0.11) to return the 11.x style of version number (presumably these have been built against XCode 12).

I verified this behavior myself. On jshell from AdoptOpenJDK (build 11.0.10+9):

```
jshell> System.getProperty("os.version");
$1 ==> "10.16"
```

With jshell from AdoptOpenJDK-11.0.11+9 (build 11.0.11+9):

```
jshell> System.getProperty("os.version");
$1 ==> "11.3"
```

References:
  * [JDK-8253702 - BigSur version number reported as 10.16, should be 11.nn](https://bugs.openjdk.java.net/browse/JDK-8253702)
  * [Big Sur is both 10.16 and 11.0 – it’s official (blog post)](https://eclecticlight.co/2020/07/21/big-sur-is-both-10-16-and-11-0-its-official/)
  * [macOS version numbering isn’t so simple (blog post)](https://eclecticlight.co/2020/08/13/macos-version-numbering-isnt-so-simple/)
  * [macOS 11.0 Big Sur: The Ars Technica review](https://arstechnica.com/gadgets/2020/11/macos-11-0-big-sur-the-ars-technica-review/) - See the third paragraph

**Comment by jthurne** [comment]
Given the findings above, we can probably let the original user who reported the issue know that `10.16` is a legitimate version number referring to MacOS Big Sur.  If they need to know which revision of Big Sur is being used, then they'll need to upgrade the JREs across all of the users and machines that can run builds to be one of the patch releases that was built with the newer version of XCode.
